charms/reactive/bus.py

```diff
@@ -444,19 +444,19 @@ def _load_module(root, filepath):
     return importlib.import_module(package + module)
 
 
+def _is_external_handler(filepath):
+    if not os.access(filepath, os.X_OK):
+        return False
+    with open(filepath, 'rb') as fp:
+        bytes = fp.read(1024)
+    if bytes.startswith(b'#!'):
+        return True
+    textchars = bytearray({7, 8, 9, 10, 12, 13, 27} | set(range(0x20, 0x100)) - {0x7f})
```

Comment:
Can this get a comment as to what exactly this check is for / why these numbers in particular? I could understand an ASCII check of `0x00-0x7F`, but not sure about these. If you're checking for a particular codepage, would it make sense to do something like this?

```python
try:
    bytes.decode('utf-8')
    return True
except UnicodeDecodeError:
    return False
```

Reply:
Yeah, though binary files should probably be considered as executables, but I also included the file extension whitelist to account for things like executable tarballs.